src/functionals.jl

```diff
@@ -30,5 +37,9 @@ end
 # end
 
 function kldivergence(P::UnivariateDistribution, Q::UnivariateDistribution)
```

Comment:
IMO it is easiest if you just change this line to
```suggestion
function kldivergence(P::Distribution{V}, Q::Distribution{V}; kwargs...) where {V<:VariateForm}
```